src/app/components/Slideshow/sliderComponents/Slider.js

```diff
@@ -0,0 +1,129 @@
+import React, { Component } from 'react';
+import { SliderLeftArrow, SliderRightArrow } from './Arrows';
+import SliderIndicator from './Indicators';
+import Slide from './Slides';
+
+import '../Slider.scss';
+
+export default class Slider extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      activeIndex: 0,
+      isStopped: false
+    };
+  }
+
+  componentDidMount = () => {
+    this.activate();
+  };
+
+  componentWillUnmount = () => {
+    this.deactivate();
+  };
+
+  activate = () => {
+    this.interval = setInterval(this.goToNextSlide, 3000);
+  };
+
+  deactivate = () => {
+    clearInterval(this.interval);
+  };
+
+  isStoppedCheck = () => {
+    const { isStopped } = this.state;
+
+    if (!isStopped) {
+      this.setState({ isStopped: true });
+    }
+  };
+
+  handleClickIndicator = id => {
+    this.goToSlide(id);
+    this.isStoppedCheck();
+  };
+
+  handleClickLeft = () => {
+    this.goToPrevSlide();
+    this.isStoppedCheck();
+  };
+
+  handleClickRight = () => {
+    this.goToNextSlide();
+    this.isStoppedCheck();
+  };
+
+  goToSlide = index => {
+    this.setState({ activeIndex: index });
+  };
+
+  goToPrevSlide = e => {
+    if (e) {
+      e.preventDefault();
+    }
+    const { activeIndex } = this.state;
+    const { slides } = this.props;
+
+    let index = activeIndex;
+    const slidesLength = slides.length;
+
+    if (index < 1) {
+      index = slidesLength;
+    }
+
+    --index;
```

Comment:
just use `index - 1` instead of unary

Reply:
Because the unary ++ and -- operators are subject to automatic semicolon insertion, differences in whitespace can change semantics of source code.